### ISSUE TITLE
cannot reproduce "unknown key value(s)" croaking

### ISSUE THREAD

**Issue by XSven** [pmqs/IO-Compress#55]
perl 5.14.2
IO::Compress 2.206

I am using this function

```
sub _gunzip_file {
  my ( $filename ) = @_;

  my $gunzip = IO::Uncompress::Gunzip->new( $filename, { Append => 1, MultiStream => 1 } );
  my $buffer = '';
  1 while $gunzip->read( $buffer ) > 0;
  die "gunzip failed: $GunzipError\n" if $GunzipError;

  return $buffer;
}
```

to gunzip files. Recently the function croaks with the message

`Compress::Raw::Zlib::Inflate::new: unknown key value(s) AppendOutput CRC32 ADLER32 at perl5/IO/Uncompress/Adapter/Inflate.pm line 34`

Unfortunately until today this happens only once and I was not able to reproduce it even with the same .gz input file that has caused the croaking.

Any idea why it could randomly happen that the key values **AppendOutput** **CRC32** **ADLER32** are treated as unknown?

**Comment by pmqs** [comment]
That is very strange. Looks like somehow the object in `$object` has got corrupted.  Without a way to reproduce it though it will be virtually impossible to fix.

If you get to a point where you can reproduce it, please report it back to me.

**Comment by XSven** [comment]
I have detected yet another non-reproducible phenomenon

`Size magic not implemented at perl5/IO/Uncompress/Base.pm line 386. `

On purpose I have not created a separate issue for that. Any idea?

**Comment by pmqs** [comment]
You are ok keeping the same ticket.

I found this reference to the "_Size magic not implemented_" error at https://github.com/Perl/perl5/issues/9701. It relates to an internal issue with Perl itself. That could also be the root-cause for the first issue you reported.

I think the only approach for fixing these issues is to upgrade to a newer version of Perl. The version you are using (5.14.2) dates from 2011. The latest version is 5.38.2.
